### ISSUE TITLE
Working on a v1 / The Adventures of Misusing Issues

### ISSUE THREAD

**Issue by Merri** [Merri/react-tabbordion#14]
Goals:

1. Both stateless and stateful versions.
2. Use `PureComponent`: remove workarounds that were made back in the React 0.13 days.
3. Improve props: merge multiple booleans to one prop, use clearer naming convention.
4. Provide separate components for dynamic/animated height of panels.
5. Add helper tools to make creation of customized components easier.
6. Maintain the great parts: WAI ARIA, styling goodness, excellent HTML/DOM structure.

Optional goals: more styles and examples, use merri.net style in demo, modernize demo app (the ES2015+ goodness), server rendered demo, provide helpers (actions/reducers) for managing state via Redux.

In short v1 will be incompatible with the 0.X series and will have performance improvements as a side effect of improving the design.

**Comment by Merri** [comment]
The v1 branch is mostly working, but I must refactor usage of `context` and rewrite tests. Also the implementation of animated panels require some thought.

**Comment by Merri** [comment]
A lot has been refined in the v1 branch: changelog and readme have been updated, docs started, some tests have been rewritten, `TabContent` now supports animation/transitions just the same way as before.
